### PR TITLE
feat: Add Sync Protocol

### PR DESCRIPTION
**Pull request by Elyniss**
Merge after: #764

### CODE REVIEW

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt [hunk 1]
```diff
@@ -1,6 +1,15 @@
...
+import com.walletconnect.android.internal.common.scope
+import com.walletconnect.android.internal.common.wcKoinApp
+import com.walletconnect.sync.client.mapper.toClient
+import com.walletconnect.sync.client.mapper.toClientError
+import com.walletconnect.sync.common.model.Events
+import com.walletconnect.sync.di.engineModule
 import com.walletconnect.sync.engine.domain.SyncEngine
+import kotlinx.coroutines.launch
 
 class SyncProtocol : SyncInterface {
```
Comment (Elyniss) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt:14]: internal will be added in following PR

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/domain/SyncEngine.kt [hunk 2]
```diff
@@ -1,5 +1,15 @@
 package com.walletconnect.sync.engine.domain
 
+import com.walletconnect.android.internal.common.model.type.EngineEvent
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+
 class SyncEngine {
```
Comment (Elyniss) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/engine/domain/SyncEngine.kt:8]: internal will be added in following PR

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt [hunk 3]
```diff
@@ -9,35 +18,71 @@ class SyncProtocol : SyncInterface {
         val instance = SyncProtocol()
     }
 
-    override fun setSyncDelegate(delegate: SyncInterface.SyncDelegate) {
-        TODO("Not yet implemented")
+    override fun initialize(params: Sync.Params.Init, onError: (Sync.Model.Error) -> Unit) {
+        try {
+            wcKoinApp.run {
+                modules(
+                    engineModule(),
```
Comment (TalhaAli00) [inline sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt:25]: let's also add a common module to add the SDK flag to user agent

**Review by TalhaAli00 — APPROVED** [review]

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt (hunk 3 above)
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt:25]: Good call!

**Review by jakubuid — APPROVED** [review]

File: sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt (hunk 3 above)
Comment (Elyniss) [reply sync/sdk/src/main/kotlin/com/walletconnect/sync/client/SyncProtocol.kt:25]: I will add in next PR, tho.

**Pull request merged**